docs/developer-docs/integrations/icrc-1/icrc1-ledger-local-setup.md

```diff
@@ -0,0 +1,291 @@
+      "wasm": "https://download.dfinity.systems/ic/d87954601e4b22972899e9957e800406a0a6b929/canisters/ic-icrc1-ledger.wasm.gz",
+      "remote": {
+        "id": {
+          "ic": "mxzaz-hqaaa-aaaar-qaada-cai"
+        }
+      }
+    }
+  },
+  "defaults": {
+    "build": {
+      "args": "",
+      "packtool": ""
+    }
+  },
+  "output_env_file": ".env",
+  "version": 1
+}
+```
+Specifying the canister id `mxzaz-hqaaa-aaaar-qaada-cai` is optional. It is set in this tutorial, so that we can be sure what we mean when referencing the deployed canister later in this tutorial.
+
+In an existing project you would only need to add the `icrc1_ledger_canister` canister to the `canisters` section.
+
+### Step 5:  Start a local replica.
+
+``` sh
+dfx start --background --clean
+```
+
+### Step 6:  Create the required identities and export initialization arguments:
+
+``` sh
+dfx identity new minter
+dfx identity use minter
+export MINTER=$(dfx identity get-principal)
+```
+Transfers from the minting account will create `Mint` transactions. Transfers to the minting account will create `Burn` transactions.
+
+Specify the token name and symbol of your choice:
+
+``` sh
+export TOKEN_NAME="My Token"
+export TOKEN_SYMBOL="XMTK"
+```
+
+Set the default identity or the identity with which you want to deploy the ledger.
+``` sh
+dfx identity use default
+export DEFAULT=$(dfx identity get-principal)
+```
+
+[OPTIONAL]
+To be able to interact and send some tokens you may want to mint some tokens when you deploy the ledger. 
+We will mint some tokens for the default identity.
+You can also specify the transfer fee for transfering tokens. 
+``` sh
+export PRE_MINTED_TOKENS=10_000_000_000
+export TRANSFER_FEE=10_000
+```
+
+The values set for archiving are the recommended values. You may change them to fit your ICRC-1 ledger's needs. 
+``` sh
+dfx identity new archive_controller
+dfx identity use archive_controller
+export ARCHIVE_CONTROLLER=$(dfx identity get-principal)
+export TRIGGER_THRESHOLD=2000
+export NUM_OF_BLOCK_TO_ARCHIVE=1000
+export CYCLE_FOR_ARCHIVE_CREATION=10000000000000
+```
+
+Check the set variables:
+
+For each variable, the exported environment variable will be used unless otherwise specified:
+-   the `TOKEN_NAME` is the human-readable name of your new token.
+-   the `TOKEN_SYMBOL` is the ticker symbol of your new token.
+-   the `MINTER` is the account of the Principal responsible for minting and burning tokens (see the [icrc-1 ledger documentation](https://github.com/dfinity/ICRC-1)).
+-   Minting 100 tokens to the `DEFAULT` (1 token is by default equal to 10^8 e8s, hence the name).
+-   Setting the transfer fee to 0.0001 tokens.
+### Step 8:  Deploy the ICRC-1 ledger canister locally:
+
+
+``` sh
+dfx deploy icrc1_ledger_canister --specified-id mxzaz-hqaaa-aaaar-qaada-cai --argument "(variant {Init = 
```

Comment:
i would remove the `--specified-id` flag here for the same reason as above, this is not a tutorial to deploy the ckBTC ledger locally

Reply:
I want to be able the canister id to be deterministic when I deploy it locally. In the coming tutorial about ICRC-1 index canisters I want to be able to reference this id. I clarified that for mainnet deployment one has to remove this flag.

Reply:
if the index canister is deployed in the same dfx project, you could also reference the ledger by `dfx canister id icrc1_ledger_canister`